internal/p2p/p2ptest/network.go

```diff
@@ -247,7 +247,7 @@ func (n *Network) MakeNode(ctx context.Context, t *testing.T, opts NodeOptions)
 	}
 
 	transport := n.memoryNetwork.CreateTransport(nodeID)
```

Comment:
Now that there is only one or none, maybe this should change instead of testing for zero.

```go
func (t *MemoryTransport) Endpoint() (Endpoint, bool) {
```

in the spirit of https://github.com/golang/go/wiki/CodeReviewComments#in-band-errors. 

Or, alternatively, I'd be fine with returning a pointer to it so we can check against `nil`. Zero checking is subtle and easily made wrong by changing fields.

Or, alternatively, I'd be fine with asserting this cannot fail, and making it panic if it can't succeed, or allowing a zero value to be valid (maybe no-op) endpoint.

Reply:
I did this (doing *Endpoint, error) somewhat reflexively.